controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1679,6 +1680,53 @@ bool master_thread::handle_cmdu_1905_topology_response(const std::string &src_ma
         }
     }
 
+    // The reported neighbors list might not be correct since the reporting al_mac hasn't received 
+    // a Topology Discovery from its neighbors yet. Therefore, remove a neighbor node only if more
+    // than 65 seconds have passed since we added this node. This promise that the reported al_mac
+    // will get the Topology Discovery messages from its neighbors and add them to the report.
+    auto last_state_change_timestamp = database.get_last_state_change(src_mac);
```

Comment:
I think this should be `al_mac` instead of `src_mac`. Normally they should be the same, but if they are not, we should actually somewhere make src_mac an alias of al_mac. That of course doesn't need to be done now, it's a much bigger change, but just use al_mac to make things consistent.

Reply:
No, the src_mac is always the al_mac of the sender.